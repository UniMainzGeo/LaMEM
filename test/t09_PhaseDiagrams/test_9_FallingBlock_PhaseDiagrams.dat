#===============================================================================
# Scaling
#===============================================================================

	units = geo

	unit_temperature = 273
	unit_length      = 10e3
	unit_viscosity   = 1e22
	unit_stress      = 1e10

#===============================================================================
# Time stepping parameters
#===============================================================================

	time_end  = 1.0   # simulation end time
	dt        = 1e-5  # time step
	dt_min    = 1e-5  # minimum time step (declare divergence if lower value is attempted)
	dt_max    = 0.1   # maximum time step
	dt_out    = 0.2   # output step (output at least at fixed time intervals)
	inc_dt    = 0.1   # time step increment per time step (fraction of unit)
	CFL       = 0.5   # CFL (Courant-Friedrichs-Lewy) criterion
	CFLMAX    = 0.5   # CFL criterion for elasticity
	nstep_max = 2     # maximum allowed number of steps (lower bound: time_end/dt_max)
	nstep_out = 1     # save output every n steps
	nstep_rdb = 0     # save restart database every n steps


#===============================================================================
# Grid & discretization parameters
#===============================================================================

# Number of cells for all segments

	nel_x = 32
	nel_y = 8
	nel_z = 32

# Coordinates of all segments (including start and end points)

	coord_x = 0.0 10.0
	coord_y = 0.0 10.0
	coord_z = -10.0 0.0

#===============================================================================
# Free surface
#===============================================================================

# Default

#===============================================================================
# Boundary conditions
#===============================================================================

# temperature on the top & bottom boundaries

	temp_top  = 0.0
	temp_bot  = 300.0;

# temperature initial guess flag
# linear profile between temp_top and temp_bot
	init_temp = 1;

#===============================================================================
# Solution parameters & controls
#===============================================================================

	gravity        = 0.0 0.0 -9.81   # gravity vector
	FSSA           = 1.0            # free surface stabilization parameter [0 - 1]
	init_guess     = 0              # initial guess flag
	eta_min        = 1e19           # viscosity upper bound
	eta_max        = 1e25           # viscosity lower limit
	DII_ref        = 1e-16          # background (reference) strain-rate

#===============================================================================
# Solver options
#===============================================================================
	SolverType 		=	direct 			# solver [direct or multigrid]
	DirectSolver 	=	mumps			# mumps/superlu_dist/pastix	
	DirectPenalty 	=	1e5

		
#===============================================================================
# Model setup & advection
#===============================================================================

	msetup         = geom              # setup type
	nmark_x        = 2                 # markers per cell in x-direction
	nmark_y        = 2                 # ...                 y-direction
	nmark_z        = 2                 # ...                 z-direction
	bg_phase       = 0                 # background phase ID


# Geometric primitives:

	<BoxStart>
		phase  = 1
		bounds = 2.5 7.5 2.5 7.5 -7.5 -2.5  # (left, right, front, back, bottom, top)
	<BoxEnd>

#===============================================================================
# Output
#===============================================================================

# Grid output options (output is always active)

	out_file_name       = FB_test # output file name
	out_pvd             = 1       # activate writing .pvd file
	out_phase           = 1
	out_density         = 1
	out_visc_total      = 1
	out_pressure        = 1
	out_temperature     = 1
	out_melt_fraction   = 1
	out_fluid_density   = 1
      
# AVD phase viewer output options (requires activation)

	out_avd     = 1 # activate AVD phase output
	out_avd_pvd = 1 # activate writing .pvd file
	out_avd_ref = 3 # AVD grid refinement factor

	

#===============================================================================
# Material phase parameters
#===============================================================================

	# Define properties of matrix
	<MaterialStart>
		ID  = 0 # phase id
		rho = 3000 # density
		disl_prof = Quarzite-Ranalli_1995
		rho_ph = TestPD
		#rho_ph_file = t9_PhaseDiagrams/
		#rho_ph_file = t9_Phase
		
	<MaterialEnd>

	# Define properties of block
	<MaterialStart>
		ID  = 1   # phase id
		rho = 2500   # density
		disl_prof = Quarzite-Ranalli_1995
		T   = 800.0  # optional temperature to set within the phase
		rho_ph = TestPD1
		#rho_ph_file = t9_PhaseDiagrams/PD_Dir1/
		rho_ph_file = PD_Dir1/
		
	<MaterialEnd>

#===============================================================================
# PETSc options
#===============================================================================

<PetscOptionsStart>

	-objects_dump

<PetscOptionsEnd>

#===============================================================================
