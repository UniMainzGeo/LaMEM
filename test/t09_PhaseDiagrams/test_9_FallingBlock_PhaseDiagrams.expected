-------------------------------------------------------------------------- 
                   Lithosphere and Mantle Evolution Model                   
     Compiled: Date: Dec  1 2020 - Time: 11:48:15 	    
-------------------------------------------------------------------------- 
        STAGGERED-GRID FINITE DIFFERENCE CANONICAL IMPLEMENTATION           
-------------------------------------------------------------------------- 
Parsing input file : ./t9_PhaseDiagrams/test_9_FallingBlock_PhaseDiagrams.dat 
   Adding PETSc option: -objects_dump
Finished parsing input file : ./t9_PhaseDiagrams/test_9_FallingBlock_PhaseDiagrams.dat 
--------------------------------------------------------------------------
Scaling parameters:
   Temperature : 273. [C/K] 
   Length      : 10000. [m] 
   Viscosity   : 1e+22 [Pa*s] 
   Stress      : 1e+10 [Pa] 
--------------------------------------------------------------------------
Time stepping parameters:
   Simulation end time          : 1. [Myr] 
   Maximum number of steps      : 2 
   Time step                    : 1e-05 [Myr] 
   Minimum time step            : 1e-05 [Myr] 
   Maximum time step            : 0.1 [Myr] 
   Time step increase factor    : 0.1 
   CFL criterion                : 0.5 
   CFLMAX (fixed time steps)    : 0.5 
   Output time step             : 0.2 [Myr] 
   Output every [n] steps       : 1 
   Output [n] initial steps     : 1 
--------------------------------------------------------------------------
--------------------------------------------------------------------------
Material parameters: 
--------------------------------------------------------------------------
   Phase ID : 0   
   dislocation creep profile: Quarzite-Ranalli_1995
   (dens)   : rho = 3000. [kg/m^3]  - Employing phase diagram: TestPD
   (disl)   : Bn = 8.63279e-20 [1/Pa^n/s]  En = 156000. [J/mol]  n = 2.4 [ ]  

   Phase ID : 1   
   dislocation creep profile: Quarzite-Ranalli_1995
   (dens)   : rho = 2500. [kg/m^3]  - Employing phase diagram: TestPD1
   (disl)   : Bn = 8.63279e-20 [1/Pa^n/s]  En = 156000. [J/mol]  n = 2.4 [ ]  
   (temp)   : T = 800. [C]  

--------------------------------------------------------------------------
--------------------------------------------------------------------------
Grid parameters:
   Total number of cpu                  : 2 
   Processor grid  [nx, ny, nz]         : [1, 1, 2]
   Fine grid cells [nx, ny, nz]         : [32, 8, 32]
   Number of cells                      :  8192
   Number of faces                      :  26112
   Maximum cell aspect ratio            :  4.00000
   Lower coordinate bounds [bx, by, bz] : [0., 0., -10.]
   Upper coordinate bounds [ex, ey, ez] : [10., 10., 0.]
--------------------------------------------------------------------------
Boundary condition parameters: 
   No-slip boundary mask [lt rt ft bk bm tp]  : 0 0 0 0 0 0 
   Top boundary temperature                   : 0. [C] 
   Bottom boundary temperature                : 300. [C] 
--------------------------------------------------------------------------
Solution parameters & controls:
   Gravity [gx, gy, gz]                    : [0., 0., -9.81] [m/s^2] 
   Surface stabilization (FSSA)            :  1. 
   Use lithostatic pressure for creep      @ 
   Minimum viscosity                       : 1e+19 [Pa*s] 
   Maximum viscosity                       : 1e+25 [Pa*s] 
   Universal gas constant                  : 8.31446 [J/mol/K] 
   Max. melt fraction (viscosity, density) : 0.15    
   Rheology iteration number               : 25  
   Rheology iteration tolerance            : 1e-06    
   Ground water level type                 : none 
--------------------------------------------------------------------------
Advection parameters:
   Advection scheme              : Euler 1-st order (basic implementation)
   Marker setup scheme           : geometric primitives
   Velocity interpolation scheme : STAG (linear)
   Marker control type           : no marker control
   Markers per cell [nx, ny, nz] : [2, 2, 2] 
   Marker distribution type      : uniform
   Background phase ID           : 0 
--------------------------------------------------------------------------
Reading geometric primitives ... done (0.000334978 sec)
--------------------------------------------------------------------------
Phase Diagrams:  
   Diagrams employed for phases  : 0, 1, 
--------------------------------------------------------------------------
Output parameters:
   Output file name                        : FB_test 
   Write .pvd file                         : yes 
   Phase                                   @ 
   Density                                 @ 
   Total effective viscosity               @ 
   Creep effective viscosity               @ 
   Velocity                                @ 
   Pressure                                @ 
   Temperature                             @ 
   Melt fraction                           @ 
   Fluid density                           @ 
--------------------------------------------------------------------------
AVD output parameters:
   Write .pvd file       : yes 
   AVD refinement factor : 3 
--------------------------------------------------------------------------
Preconditioner parameters: 
   Matrix type                   : monolithic
   Penalty parameter (pgamma)    : 1.000000e+05
   Preconditioner type           : user-defined
--------------------------------------------------------------------------
Solver parameters specified: 
   Outermost Krylov solver       : gmres 
   Solver type                   : parallel direct/lu 
   Solver package                : mumps 
--------------------------------------------------------------------------
Saving output ... done (0.0714929 sec)
--------------------------------------------------------------------------
================================= STEP 1 =================================
--------------------------------------------------------------------------
Current time        : 0.00000 [Myr] 
Tentative time step : 0.00001 [Myr] 
--------------------------------------------------------------------------
  0 SNES Function norm 2.332692371613e+00 
  0 PICARD ||F||/||F0||=1.000000e+00 
  Linear js_ solve converged due to CONVERGED_RTOL iterations 1
  1 SNES Function norm 5.571912044213e-02 
  1 PICARD ||F||/||F0||=2.388618e-02 
  Linear js_ solve converged due to CONVERGED_RTOL iterations 2
  2 SNES Function norm 1.651339576622e-02 
  2 MMFD   ||F||/||F0||=7.079114e-03 
  Linear js_ solve converged due to CONVERGED_RTOL iterations 6
  3 SNES Function norm 5.209263956390e-03 
  3 MMFD   ||F||/||F0||=2.233155e-03 
  Linear js_ solve converged due to CONVERGED_RTOL iterations 7
  4 SNES Function norm 4.658608054775e-04 
  4 MMFD   ||F||/||F0||=1.997095e-04 
  Linear js_ solve converged due to CONVERGED_RTOL iterations 8
  5 SNES Function norm 3.025355901525e-04 
  5 MMFD   ||F||/||F0||=1.296937e-04 
  Linear js_ solve converged due to CONVERGED_RTOL iterations 9
  6 SNES Function norm 2.292297534939e-04 
--------------------------------------------------------------------------
SNES Convergence Reason : ||F|| < rtol*||F_initial|| 
Number of iterations    : 6
SNES solution time      : 16.1469 (sec)
--------------------------------------------------------------------------
Residual summary: 
   Continuity: 
      |Div|_inf = 8.305482302629e-18 
      |Div|_2   = 5.844071911041e-17 
   Momentum: 
      |mRes|_2  = 2.292297534939e-04 
--------------------------------------------------------------------------
Actual time step : 0.00001 [Myr] 
--------------------------------------------------------------------------
Saving output ... done (0.0684841 sec)
--------------------------------------------------------------------------
================================= STEP 2 =================================
--------------------------------------------------------------------------
Current time        : 0.00001 [Myr] 
Tentative time step : 0.00001 [Myr] 
--------------------------------------------------------------------------
  0 SNES Function norm 2.292297535091e-04 
  0 PICARD ||F||/||F0||=1.000000e+00 
  Linear js_ solve converged due to CONVERGED_RTOL iterations 2
  1 SNES Function norm 8.000601248103e-05 
  1 PICARD ||F||/||F0||=3.490211e-01 
  Linear js_ solve converged due to CONVERGED_RTOL iterations 2
  2 SNES Function norm 1.457394460255e-05 
  2 PICARD ||F||/||F0||=6.357789e-02 
  Linear js_ solve converged due to CONVERGED_RTOL iterations 2
  3 SNES Function norm 1.238883010322e-06 
  3 MMFD   ||F||/||F0||=5.404547e-03 
  Linear js_ solve converged due to CONVERGED_RTOL iterations 8
  4 SNES Function norm 4.129666607849e-08 
--------------------------------------------------------------------------
SNES Convergence Reason : ||F|| < atol 
Number of iterations    : 4
SNES solution time      : 8.42341 (sec)
--------------------------------------------------------------------------
Residual summary: 
   Continuity: 
      |Div|_inf = 1.370606842137e-18 
      |Div|_2   = 4.915173886604e-18 
   Momentum: 
      |mRes|_2  = 4.129666607849e-08 
--------------------------------------------------------------------------
Actual time step : 0.00001 [Myr] 
--------------------------------------------------------------------------
Saving output ... done (0.072412 sec)
--------------------------------------------------------------------------
=========================== SOLUTION IS DONE! ============================
--------------------------------------------------------------------------
Total solution time : 24.9097 (sec) 
--------------------------------------------------------------------------
